morpho/processors/diagnostics/CalibrationProcessor.py

```diff
@@ -0,0 +1,278 @@
+'''
+Processor for calibrating results; i.e., determining how often posteriors are consistent with "true" values assumed to generate fake data.
+Useful for sensitivity analyses.
+
+Authors: T. E. Weiss
+Date: May 2020
+'''
+
+from __future__ import absolute_import
+
+import numpy as np
+import math
+
+from morpho.utilities import morphologging, reader
+from morpho.processors import BaseProcessor
+from morpho.processors.IO import IOROOTProcessor
+logger = morphologging.getLogger(__name__)
+
+__all__ = []
+__all__.append(__name__)
+
+
+class CalibrationProcessor(BaseProcessor):
+    '''
+    Performs a Bayesian sensitivity calibration for a continuous parameter - i.e., computes the coverage of a credible interval. Uses either an upper limit or upper and lower bounds on a posterior, depending on user input. Prints the coverage as well as (optionally) the median and mean credible windows.
+    
+    Required input:
+        files: List of strings naming ROOT files produced by an ensemble of morpho runs.
+        in_param_names: List of strings naming parameters of interest inputted to the generator.
+        
+    Optional input:
+        cred_interval: List with float elements between 0 and 1 defining a posterior credible window; defaults to [0.05, 0.95]. If len(cred_interval)==1, this procesor finds the coverage of a limit. If len(cred_interval)==2, it finds the coverage of an interval.
+        root_in_tree: Tree containing data generation input values in each file. Defaults to "input".
+        root_post_tree: Tree containing analysis posteriors. Defaults to "analysis".
+        post_param_names: List of strings naming posteriors produced by Stan analysis for a parameters of interest. Defaults to self.in_param_names.
+        verbose: If True, print information for each pseudo-experiment and a summary at the end. If False, only print the summary.
+        quantile: If True, compute quantile credible intervals. Otherwise, compute highest density intervals.
+        check_if_nonzero: If True, check whether posteriors allow the parameters to be distinguished from zero (given some credible interval).
+        
+    Results:
+        coverages: dictionary containing coverage of interval given by self.cred_interval, for each parameter in self.in_param_names
+    '''
+    def InternalConfigure(self,params):
+        #Required input
+        self.files = reader.read_param(params,'files','required')
+        self.in_param_names = reader.read_param(params,'in_param_names','required')
```

Comment:
Maybe check the files exists before leaving the Configure part: simply add a
```python
for file in self.files:
    if not os.path.exists(file):
        logger.error("File {} doesn't exist".format(file))
        return False
```
Add a `import os` too...

Reply:
Unless the files are created by another processor before this one, in which case, we can just print a logger.warning instead of logger.error and drop the `return False`

Reply:
Took option 2 by adding a warning:
```python
for file in self.files:
    if not exists(file):
         logger.warning("File {} doesn't exist".format(file))
```

Reply:
Very good idea, thank you for doing this.